File: 240/Go/main.go
package main

import "fmt"

func searchMatrix(matrix [][]int, target int) bool {
	for _, v := range matrix {
		for _, val := range v {
			if val == target {
				return true
			}
		}
	}
	return false
}
func main() {
	matrix := [][]int{
		{1, 4, 7, 11, 15},
		{2, 5, 8, 12, 19},
		{3, 6, 9, 16, 22},
		{10, 13, 14, 17, 24},
		{18, 21, 23, 26, 30},
	}
	fmt.Println(searchMatrix(matrix, 5))
	matrix = [][]int{
		{-1, 3},
	}
	fmt.Println(searchMatrix(matrix, 3))
}

File: 240/Cpp/main.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        for ( const auto& row : matrix ) {
            auto it = lower_bound(row.begin(), row.end(), target);
            if (it != row.end() && *it == target ) {
                return true;
            }
        }
        return false;
    }
};

int main () {
    vector< vector<int> > nums = {
        {1, 4, 7, 11, 15},
        {2, 5, 8, 12, 19},
        {3, 6, 9, 16, 22},
        {10, 13, 14, 17, 24},
        {18, 21, 23, 26, 30},
    };
    Solution s;
    cout << s.searchMatrix(nums, 5) << endl;
}